### PR TITLE
Fixes for 0.7

### PR DESCRIPTION
**Pull request by rsrock**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,7 @@ os:
     - linux
     - osx
 julia:
-    - 0.4
-    - 0.5
+    - 0.7
     - nightly
```
Comment (bjarthur) [inline .travis.yml:7]: the only real difference with #11 is that i allowed failures on nightly to make hexagons like gadfly and compose.

File: REQUIRE [hunk 2]
```diff
@@ -1,2 +1,2 @@
-julia 0.4
+julia 0.7
 Compat 0.17
```
Comment (bjarthur) [inline REQUIRE:2]: are their any remaining calls to @compat ?  if not can we remove this requirement?

File: src/Hexagons.jl [hunk 3]
```diff
@@ -123,22 +119,21 @@ const CUBIC_HEX_NEIGHBOR_OFFSETS = [
 neighbors(hex::Hexagon) = HexagonNeighborIterator(convert(HexagonCubic, hex))
 
 length(::HexagonNeighborIterator) = 6
-start(::HexagonNeighborIterator) = 1
-done(::HexagonNeighborIterator, state::Int) = state > 6
 
-function next(it::HexagonNeighborIterator, state::Int)
+function Base.iterate(it::HexagonNeighborIterator, state=1)
```
Comment (bjarthur) [inline src/Hexagons.jl:123]: here and elsewhere, must one qualify `iterate` with `Base.` ?

File: src/Hexagons.jl [hunk 4]
```diff
@@ -154,37 +149,36 @@ const CUBIC_HEX_DIAGONAL_OFFSETS = [
...
     x_center::Float64
     y_center::Float64
     xsize::Float64
     ysize::Float64
 
     function HexagonVertexIterator(x, y, xsize=1.0, ysize=1.0)
-        new((@compat Float64(x)), (@compat Float64(y)),
-            (@compat Float64(xsize)), (@compat Float64(ysize)))
+        new((Float64(x)), (Float64(y)),
+            (Float64(xsize)), (Float64(ysize)))
```
Comment (bjarthur) [inline src/Hexagons.jl:174]: here and elsewhere, there are extra parens that could be removed

File: src/Hexagons.jl [hunk 5]
```diff
@@ -291,17 +285,18 @@ end
...
 end
-done(it::HexagonSpiralIterator, state::HexagonSpiralIteratorState) = state.hexring_i > it.n
 
 # The state of a HexagonSpiralIterator consists of
 # 1. an Int, the index of the current ring
 # 2. a HexagonRingIterator and its state to keep track of the current position
 #    in the ring.
-function next(it::HexagonSpiralIterator, state::HexagonSpiralIteratorState)
+function Base.iterate(it::HexagonSpiralIterator,
+                      state::HexagonSpiralIteratorState=_start(it))
```
Comment (bjarthur) [inline src/Hexagons.jl:298]: are their pros and cons to calling `_start` in the function signature, versus creating a second `iterate` method with a single input argument as i did in #11 ?  the latter is easier to read to me, but i'm not familiar enough with julia internals to know if there's a performance difference.

**Review by bjarthur — APPROVED** [review]
looks great!  next time maybe ping me or @tlnagy if you don't get a timely response.  what do you use hexagons.jl for?  i guess i'll chock up my work on #11 to an exercise in learning the new iterate protocol!

**Comment by rsrock** [comment]
No worries @bjarthur! I don't use Hexagons for anything other than Gadfly.  Also a good chance to learn the new iteration protocol myself 😄.

I will go over your comments in the next few days

**Comment by bjarthur** [comment]
great!  we're hoping to release a beta gadfly for field testing in the next few days too.  this PR is critical to that plan.

**Comment by tlnagy** [comment]
Now that the monster Compose PR is merged (https://github.com/GiovineItalia/Compose.jl/pull/282). This PR is high priority for getting Gadfly working. @rsrock do you have any time to look at this in the next couple of days?

@bjarthur might be worth re-opening your PR if we don't hear back from Ron. I'm fine with either iteration implementation.

**Pull request merged**